src/script/interpreter.cpp

```diff
@@ -1434,16 +1434,23 @@ bool GenericTransactionSignatureChecker<T>::CheckLockTime(const CScriptNum& nLoc
     if (nLockTime > (int64_t)txTo->nLockTime)
         return false;
 
-    // Finally the nLockTime feature can be disabled and thus
-    // CHECKLOCKTIMEVERIFY bypassed if every txin has been
-    // finalized by setting nSequence to maxint. The
+    // Finally the nLockTime feature can be disabled in IsFinalTx()
+    // and thus CHECKLOCKTIMEVERIFY bypassed if every txin has
+    // been finalized by setting nSequence to maxint. The
     // transaction would be allowed into the blockchain, making
     // the opcode ineffective.
     //
     // Testing if this vin is not final is sufficient to
     // prevent this condition. Alternatively we could test all
     // inputs, but testing just this input minimizes the data
     // required to prove correct CHECKLOCKTIMEVERIFY execution.
+    //
+    // This additional constraint enables a kind of lightweight
+    // covenant: by using OP_CHECKLOCKTIMEVERIFY an output may
+    // be unspendable until a specific nBlockHeight/nBlockTime
+    // committed in the scriptPubKey, as opposed to a spending
+    // transaction's nLockTime only being constrained by
+    // signature verification.
```

Comment:
I think the Bitcoin Core source code (or interpreter source code) is the wrong place to document what is possible with each opcode. Better places to document this is the BIPs repo (or stackexchange, the mailing list, your private blog or website, ...).

In fact the BIP already mentions several use cases. See https://github.com/bitcoin/bips/blob/master/bip-0065.mediawiki#Motivation . Should the other use-cases be copy-pasted here as well? If a new use-case is found, should it be copied here, too? How much review/maintenance time is this going to consume?

Also, you mention the word `covenant`, which isn't defined in our source code, nor the BIP text. I think this section adds more confusion for anyone reading the BIP and then this section.

Reply:
That sounds reasonable to me and fits in with the original motivation. @ariard since this was your [suggestion](https://github.com/bitcoin/bitcoin/pull/18096#discussion_r379129481) do you mind if I remove it?

Reply:
I went ahead and removed this addition.

Reply:
Fine to remove it. IIRC, I did advise you this comment because I was surprised to see nowhere this relation between nLocktime enforcement and finality of inputs documented. At least not in BOLT3, where timeout transactions nSequence are carefully set to non-final.

Overall I agree with @MarcoFalke. Though I would say some mechanism usage towards specific Bitcoin applications are already documented in the codebase. For e.g the carve-out rule in the mempool : https://github.com/bitcoin/bitcoin/blob/683dda2a70e7a210996fa34be23bd0c563971ba9/src/validation.cpp#L801